workers/unity/Assets/Playground/Scripts/MonoBehaviours/FlashOnCollision.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Generated.Improbable.Gdk.Tests.ComponentsWithNoFields;
+using Generated.Playground;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+
+public class FlashOnCollision : MonoBehaviour
+{
+    [Require] private Generated.Playground.Collisions.Reader reader;
+
+    private float collideTime;
+    private bool flashing = false;
+
+    void OnEnable()
+    {
+        if (reader != null)
+        {
+            reader.OnPlayerCollided += HandleCollisionEvent;
+        }
+    }
+
+    void OnDisable()
+    {
+        if (reader != null)
+        {
+            reader.OnPlayerCollided -= HandleCollisionEvent;
+        }
+    }
+
+    void HandleCollisionEvent(PlayerCollidedEvent e)
+    {
+        collideTime = Time.time;
+        flashing = true;
+        var renderer = gameObject.GetComponent<MeshRenderer>();
+        renderer.material.SetColor("_Color", UnityEngine.Color.red);
+    }
+
+    void Update()
+    {
+        if (flashing && Time.time - collideTime > 0.2f)
```

Comment:
supernit: magic number can be turned into a const var

Reply:
Since it's a monobehaviour, why not make it a property you can change in the editor! Same with the flash color above too.